Development

samwachspress opened the pull request:
Northwestern MTI release

danielecook commented on base/templates/admin/data_edit.html:
```diff
@@ -0,0 +1,239 @@
+        <div class='row'>
+          <div class='col-md-4'> 
+            <strong> Dataset: </strong>
+          </div> <!-- /col-md-4 -->
```
If you use `{# #}` these will be removed before being sent to the user. Not a big deal really just FYI.

danielecook commented on base/templates/admin/data_list.html:
```diff
@@ -0,0 +1,56 @@
+
+<div class="row">
+  <div class="col-md-12">
+    <table>
```
You can format this and make it look a lot nicer by adding `class="table"`

danielecook commented on base/templates/admin/users_list.html:
```diff
@@ -0,0 +1,53 @@
+
+<div class="row">
+  <div class="col-md-12">
+    <table>
```
Consider adding `class="table"`

danielecook commented on base/templates/download.html:
```diff
@@ -0,0 +1,20 @@
+
+<script>
+$(document).ready(function() {
+  $('#download-link').get(0).click();
```
This is probably a good idea. There is a 'download_bams_script.sh' page that no longer works:

https://development-1-6-0-dot-andersen-lab.appspot.com/data/release/20210121/download/download_strain_bams.sh

We can either (1) have users request access or (2) allow users to download with a 'requester pays' option setup on GCP. This would allow users to download the data but they would have to pay for the download costs.

danielecook reviewed (approved):
I would fix the download_all_bams.sh script before launching this.

samwachspress committed:
write bam_bai download script to cloud bucket

samwachspress committed:
Merge pull request #25 from northwestern-mti/feature/bg-thread

write bam_bai download script to cloud bucket

samwachspress committed:
fix blobPath

samwachspress committed:
Merge pull request #26 from northwestern-mti/feature/bg-thread

fix blobPath

samwachspress committed:
update indel primer

samwachspress committed:
update queue

samwachspress committed:
fix dockerfile

samwachspress committed:
Merge pull request #27 from northwestern-mti/feature/indel-primer

Feature/indel primer

samwachspress committed:
fix capitalization

samwachspress committed:
secure indel primer

samwachspress committed:
noindex, update indel primer

samwachspress committed:
Merge pull request #28 from northwestern-mti/feature/indel-primer

Feature/indel primer

samwachspress committed:
remove jwt-csrf and revise text content

samwachspress committed:
Merge pull request #29 from northwestern-mti/feature/indel-primer

remove jwt-csrf and revise text content

samwachspress committed:
update decrypt key

samwachspress committed:
Merge pull request #30 from northwestern-mti/feature/indel-primer

update decrypt key

samwachspress committed:
do not save output of vk primer

samwachspress committed:
log exception

samwachspress committed:
remove none string cast

samwachspress committed:
Merge pull request #31 from northwestern-mti/hotfix/downloads

log exception

samwachspress committed:
use service account to init bucket client

samwachspress committed:
Merge pull request #32 from northwestern-mti/hotfix/downloads

use service account to init bucket client

samwachspress committed:
fix script download path

samwachspress committed:
fix search download tab

samwachspress committed:
Merge pull request #33 from northwestern-mti/hotfix/downloads

Hotfix/downloads

samwachspress committed:
do not save output of vk primer

samwachspress committed:
Merge branch 'feature/indel-primer' of https://github.com/northwestern-mti/CeNDR-1 into feature/indel-primer

samwachspress committed:
update indel primer reports

samwachspress committed:
update text

samwachspress committed:
Merge pull request #34 from northwestern-mti/feature/indel-primer

Feature/indel primer

samwachspress committed:
fix snackbar colors

samwachspress committed:
update cron

samwachspress committed:
update content

samwachspress committed:
ssl exempt tasks url

samwachspress committed:
verify ip for cron tasks

samwachspress committed:
update browser, buttons, ip results

samwachspress committed:
update shibboleth sp x509 cert

samwachspress committed:
fix hash id task queueing

samwachspress committed:
update protocols

samwachspress committed:
handle task queue exception

samwachspress committed:
whitespace

samwachspress committed:
scroll lock during paste or sample load

samwachspress committed:
ip text update

samwachspress committed:
fix igv browser

samwachspress committed:
fix igv browser

samwachspress committed:
Merge pull request #35 from northwestern-mti/feature/genome-browser

fix igv browser

samwachspress committed:
log cron task ip

samwachspress committed:
Merge branch 'development' of https://github.com/northwestern-mti/CeNDR-1 into development

samwachspress committed:
temp disable ip checking

samwachspress committed:
ignore ip, x-* headers are stripped from external req

samwachspress committed:
remove comment

samwachspress committed:
improve indel primer list page

samwachspress committed:
newline at eof

samwachspress committed:
Merge pull request #36 from northwestern-mti/feature/indel-primer

improve indel primer list page

samwachspress committed:
fix bam tracks loading

samwachspress committed:
remove variants

samwachspress committed:
btn

samwachspress committed:
Merge pull request #37 from northwestern-mti/feature/indel-primer

Feature/indel primer

samwachspress committed:
sample tracks

samwachspress committed:
Merge pull request #38 from northwestern-mti/feature/indel-primer

sample tracks

samwachspress committed:
updates to igv browser

samwachspress committed:
Merge pull request #39 from northwestern-mti/feature/indel-primer

updates to igv browser

samwachspress committed:
improve datatables

samwachspress committed:
eof newline

samwachspress committed:
Merge pull request #40 from northwestern-mti/feature/heritability

improve datatables

samwachspress committed:
initial commit

samwachspress committed:
fix task queue properties

samwachspress committed:
update retry logic for task queues

samwachspress committed:
add dockerfile

samwachspress committed:
replace csi links with tbi

samwachspress committed:
update queue logic

samwachspress committed:
Update README.md

samwachspress committed:
Merge pull request #42 from northwestern-mti/feature/csi-tbi

replace csi links with tbi

samwachspress committed:
update gene browser, relocate

samwachspress committed:
cleanup wb versions, gbrowser, add vbrowser

samwachspress committed:
construct variant db schema and etl fn

samwachspress committed:
update browsers

samwachspress committed:
Merge pull request #43 from northwestern-mti/feature/variant-browser

Feature/variant browser

samwachspress committed:
remove cloud run containers

samwachspress committed:
remove junk env_configs

samwachspress committed:
migrate to external postgresql db

samwachspress committed:
Merge pull request #44 from northwestern-mti/postgres

Postgres

samwachspress committed:
update dockerfile

samwachspress committed:
Merge pull request #45 from northwestern-mti/postgres

update dockerfile

samwachspress committed:
update requirements

samwachspress committed:
update db uri

samwachspress committed:
add cloud_sql_instance unix socket to app engine

samwachspress committed:
update connection string for unix socket

samwachspress committed:
fix app.yaml

samwachspress committed:
move cloud sql init to start of app.yaml

samwachspress committed:
update env_conf

samwachspress committed:
fix config

samwachspress committed:
update env_conf and config to force code reload

samwachspress committed:
force relod

samwachspress committed:
Merge pull request #46 from northwestern-mti/feature/force-reload

force reload

samwachspress committed:
again

samwachspress committed:
Merge pull request #47 from northwestern-mti/feature/force-reload

again

samwachspress committed:
update sql query for postgres

samwachspress committed:
Merge pull request #48 from northwestern-mti/feature/force-reload

update sql query for postgres

samwachspress committed:
fix isotype count query

samwachspress committed:
Merge pull request #49 from northwestern-mti/feature/force-reload

fix isotype count query

samwachspress committed:
update navbar text

samwachspress committed:
Merge pull request #50 from northwestern-mti/feature/force-reload

update navbar text

samwachspress committed:
travis ver

samwachspress committed:
Merge branch 'development' of https://github.com/northwestern-mti/CeNDR-1 into development

samwachspress committed:
update psql uri

samwachspress committed:
add to_json for wormbase gene summary

samwachspress committed:
jsonify gene search results

samwachspress committed:
fix homolog search and to_json

samwachspress committed:
return more homolog results

samwachspress committed:
col reordering and validation

samwachspress committed:
Merge pull request #51 from northwestern-mti/feature/variant-browser

col reordering and validation

samwachspress committed:
use name to ref columns

samwachspress committed:
Merge pull request #52 from northwestern-mti/feature/variant-browser

use name to ref columns

samwachspress committed:
redraw highlight on page change

samwachspress committed:
Merge pull request #53 from northwestern-mti/feature/variant-browser

redraw highlight on page change

samwachspress committed:
ref strains col by name not index

samwachspress committed:
Merge pull request #54 from northwestern-mti/feature/variant-browser

ref strains col by name not index

samwachspress committed:
block empty form submit

samwachspress committed:
ignore case for gene searches

samwachspress committed:
update db user and email CCs

samwachspress committed:
Merge pull request #55 from northwestern-mti/feature/variant-browser

ignore case for gene searches

samwachspress committed:
layout updates with bootstrap

samwachspress committed:
Merge pull request #56 from northwestern-mti/feature/flexbox

layout updates with bootstrap

samwachspress committed:
fix buttons

samwachspress committed:
stylesheet

samwachspress committed:
Merge pull request #57 from northwestern-mti/feature/flexbox

fix buttons

samwachspress committed:
fix other btn

samwachspress committed:
Merge pull request #58 from northwestern-mti/feature/flexbox

fix other btn

samwachspress committed:
add styles update

samwachspress committed:
Merge pull request #59 from northwestern-mti/feature/flexbox

add styles update

samwachspress committed:
travis++

samwachspress committed:
Merge pull request #60 from northwestern-mti/feature/flexbox

travis++

samwachspress committed:
add nemascan task queue

samwachspress committed:
remove logging

samwachspress committed:
file upload form stub

samwachspress committed:
add nemascan task datastore kind

samwachspress committed:
add gcloud api to upload file objects as blobs

samwachspress committed:
create mapping file upload page

samwachspress committed:
index for h2 task should be uuid not data hash

samwachspress committed:
Update DB creation and add variant annotation

samwachspress committed:
update cloud sql db names

samwachspress committed:
add readme

samwachspress committed:
missing space

samwachspress committed:
create timing decorator

samwachspress committed:
move constants

samwachspress committed:
Merge pull request #62 from northwestern-mti/feature/heritability

index for h2 task should be uuid not data hash

samwachspress committed:
Merge pull request #63 from northwestern-mti/feature/variant-browser

Feature/variant browser

samwachspress committed:
Merge branch 'development' into feature/nemascan

samwachspress committed:
remove sqlite_path to surface any ref errors

samwachspress committed:
whitespace

samwachspress committed:
update mapping text prompt

samwachspress committed:
add list_files to gcloud api

samwachspress committed:
initial commit for mapping result pages (needs work)

samwachspress committed:
mapping view updates

samwachspress committed:
whitespace

samwachspress committed:
Merge pull request #61 from northwestern-mti/feature/nemascan

Feature/nemascan

samwachspress committed:
update env config

samwachspress committed:
Merge pull request #70 from northwestern-mti/feature/nemascan

update env config

samwachspress committed:
small fixes

samwachspress committed:
Merge pull request #71 from northwestern-mti/feature/nemascan

small fixes

samwachspress committed:
remove filename from schema since it is standardized

samwachspress committed:
hash file contents instead of file object

samwachspress committed:
stage uploaded file on server for hashing

samwachspress committed:
whitespace

samwachspress committed:
Merge pull request #72 from northwestern-mti/feature/nemascan

Feature/nemascan

samwachspress committed:
fix file hashing

samwachspress committed:
Merge pull request #73 from northwestern-mti/feature/nemascan

fix file hashing

samwachspress committed:
fix button colors, update queue

samwachspress committed:
fix warning snackbar readability

samwachspress committed:
add mapping results pages

samwachspress committed:
link profile to mapping results page

samwachspress committed:
add link to partial result filelist

samwachspress committed:
cleanup checks for existing data, results; switch task id from data_hash to uuid because it makes testing painful

samwachspress committed:
consistency

samwachspress committed:
Merge pull request #74 from northwestern-mti/feature/nemascan

Feature/nemascan

samwachspress committed:
update mapping result views

samwachspress committed:
Merge pull request #75 from northwestern-mti/feature/nemascan

update mapping result views

samwachspress committed:
update some styling

samwachspress committed:
Merge pull request #76 from northwestern-mti/feature/nemascan

update some styling

samwachspress committed:
update checkout button color

samwachspress committed:
enable strain sets 7 and 8

samwachspress committed:
add strains 9 and 10 as disabled

samwachspress committed:
fix dict -> json conversion

samwachspress committed:
Merge pull request #77 from northwestern-mti/hotfix/bugsmash

Hotfix/bugsmash

samwachspress committed:
remove legacy code

samwachspress committed:
convert json

samwachspress committed:
Merge pull request #78 from northwestern-mti/hotfix/bugsmash

Hotfix/bugsmash

samwachspress committed:
fix template path

samwachspress committed:
fix about page (dict -> json)

samwachspress committed:
alert user for file error, dont store datastore entry

samwachspress committed:
Merge pull request #79 from northwestern-mti/hotfix/bugsmash

Hotfix/bugsmash

samwachspress committed:
mapping fixes

samwachspress committed:
Merge pull request #80 from northwestern-mti/hotfix/bugsmash

mapping fixes

samwachspress committed:
fix homepage layout issue

samwachspress committed:
add variant impact filter

samwachspress committed:
Merge pull request #81 from northwestern-mti/hotfix/bugsmash

Hotfix/bugsmash

samwachspress committed:
style some buttons, update Form -> FlaskForm

samwachspress committed:
add vbrowser feature requests

samwachspress committed:
Merge pull request #82 from northwestern-mti/hotfix/bugsmash

Hotfix/bugsmash

samwachspress committed:
fetch isotype values to populate checklist in variant browser

samwachspress committed:
disable cloud debugger

samwachspress committed:
update mapping status info

samwachspress committed:
add landing page tiles

samwachspress committed:
fix button styles and links from home page to landing pages

samwachspress committed:
remove inconsistent period

samwachspress committed:
Merge pull request #83 from northwestern-mti/hotfix/bugsmash

Hotfix/bugsmash

samwachspress committed:
last bits

samwachspress committed:
travis ver

samwachspress committed:
fix travis

samwachspress committed:
force cache busting of css

samwachspress committed:
disable NU sso

samwachspress committed:
update mapping instruction text

samwachspress committed:
add alt text to homepage video

samwachspress committed:
fix release notes

samwachspress committed:
travis ver

samwachspress committed:
fix link styles, update css version

katiesevans committed:
add point estimate if infinite

samwachspress committed:
Merge pull request #1 from AndersenLab/fix_infinite_ci

add point estimate if infinite

samwachspress committed:
add results download

samwachspress committed:
update NU styles

samwachspress committed:
travis++

samwachspress committed:
force clear css cache

samwachspress committed:
travis++

samwachspress committed:
force tls ver

samwachspress committed:
fix soft filter isotype vcf gz tbi

samwachspress committed:
add default cron task

samwachspress committed:
update travis version

samwachspress committed:
save to bam path

samwachspress committed:
travis++

samwachspress committed:
update version and fix bam download

samwachspress committed:
Merge pull request #85 from northwestern-mti/version-1.7.0g

Version 1.7.0g

samwachspress committed:
restore dev env settings

samwachspress committed:
sync env.yaml from local

samwachspress committed:
Initial commit

samwachspress committed:
migrate cloud function to separate repo

samwachspress committed:
Add 'external/cloud-functions/generate-thumbnails/' from commit 'c3bb8f4aa406b4be8d3617d8858b2f73ad0583e1'

git-subtree-dir: external/cloud-functions/generate-thumbnails
git-subtree-mainline: 3da27d3fd4d5839f7bb8106733dfef304278a252
git-subtree-split: c3bb8f4aa406b4be8d3617d8858b2f73ad0583e1

samwachspress committed:
Merge pull request #86 from northwestern-mti/feature/git-submodule

Feature/git submodule

samwachspress committed:
remove internal mapping_worker prototype code

samwachspress committed:
remove internal cloud_function code, moved to external repo

samwachspress committed:
make env_config path static instead of relative

samwachspress committed:
Update conrtributing markdown

samwachspress committed:
minimal conda environment (exclude resolved pkgs)

samwachspress committed:
remove obsolete conda env

samwachspress committed:
add default cloud config.txt

samwachspress committed:
Add 'external/cloud-run/h2calc/' from commit '3802b4d9775afc870fd5d8464227ba393dbbd082'

git-subtree-dir: external/cloud-run/h2calc
git-subtree-mainline: 6809725c42ce57d333f8e76a4c7a8520ada32517
git-subtree-split: 3802b4d9775afc870fd5d8464227ba393dbbd082

samwachspress committed:
link external modules

samwachspress committed:
add docker images as submodules

samwachspress committed:
add submodules

samwachspress committed:
Merge pull request #87 from northwestern-mti/cleanup

Cleanup
